uint8 type
float32 x
float32 y
uint32 nodeId
